__tests__/symbols.test.js

```diff
@@ -18,6 +18,11 @@ describe('getVariables()', () => {
     ['no whitespace', 'a+b', ['a', 'b']],
     ['same variable multiple times', 'a * b + a', ['a', 'b']],
     ['all operators', '(a + -b) / ((2 * c) - 3 x d)', ['a', 'b', 'c', 'd']],
+    ['formula includes functions', 'max(1, 2, a)', ['a']],
+    ['variable with function name', 'max(1, 2, a, max)', ['a', 'max']],
+    // TODO: The below fail (should they?)
+    //['TODO -fails', 'x-1', ['x']], // fix this in separate pr
+    //['TODO -fails', 'x - 1', ['x']], // fix this in separate pr
```

Comment:
Is it ok that 'x' and 'u' can't be used as variables?

Reply:
I'm not sure that it's worth it for the complexity and maintainability, but I did come up with some regexes in order to detect if x/u is in the string as a variable. What do you think?

Regex to detect the var x: https://regex101.com/r/JmUKaa/9
Regex to detect the var u: https://regex101.com/r/JmUKaa/10

Reply:
That's one way to go, the other one would be

* do not treat `x` as a possible multiplication symbol
* use a different symbol for the unary operator - one that cannot appear in a variable name

Reply:
That's a simpler option for sure! Love the second point! However the first point (I believe) would be a non-trivial loss in functionality (and both would require a new major version if I'm not mistaken). 

For the second point, what about `~`? (because it looks like a minus and I can't think of a mathematical operation which uses it)

Reply:
About the first point, it's fine then - let's create a new major version. I find no reason to keep supporting `x` as a multiplication symbol when `*` does the same job and it's the symbol used by programming languages too.

@mcccclean are you happy for us to drop support for `x` as a multiplication symbol, cause it messes up with variable substitution if a variable is called `x`?

Reply:
Also, I was under the impression that `u` is the internal representation of the unary minus symbol, used to distinguish the unary from non-unary `-`. Again, we don't have to expose this to our users, so any symbol (such as `~` which I like) will do

Reply:
>cause it messes up with variable substitution if a variable is called `x`

To be clear, I don't believe that this is a technical impossibility, I just didn't take the time to solve it today. And also I believe that we should keep non-devs in mind, who will not be familiar with `*` meaning multiplication. (I'm not strongly against dropping support, just wanted to lay out the counterarguments)

> I was under the impression that u is the internal representation of the unary minus symbol

That's correct, I didn't consider that ALL the `u`s will be replaced, so there is no issue there. My bad 👍.

> any symbol (such as ~ which I like) will do

There is also no need to change the symbol for now (equally no reason **not** to change it)

Reply:
I agree about keeping non-devs in mind. I believe that `*` is a symbol that non-devs of our time are pretty familiar with, see: Excel formulas.

Another approach I just thought of : we could use × instead of x. But again, this is overengineering in my opinion. I suggest just supporting `*`, I'm pretty sure our data stewards will have to deal with much more complexity than this.

Reply:
> Excel formulas.

True

> I'm pretty sure our data stewards will have to deal with much more complexity than this.

Also very true.

I'm happy if McLean is.